snap/info_snap_yaml_test.go

```diff
@@ -1398,6 +1398,21 @@ apps:
 	})
 }
 
+func (s *YamlSuite) TestDaemonInvalidSocketMode(c *C) {
+	y := []byte(`name: wat
+version: 42
+apps:
+ svc:
+   command: svc
+   sockets:
+     sock:
+       listen-stream: 8080
+       socket-mode: asdfasdf
+`)
+	_, err := snap.InfoFromSnapYaml(y)
+	c.Assert(err, NotNil)
```

Comment:
Nitpick: use `ErrorMatches` to highlight what this test is really checking.

Reply:
I compared the whole message, since I couldn't get the regexp to work